lib/screens/explore_screens/vote_screens/vote/interactor/viewmodels/proposal_type_model.dart

```diff
@@ -17,37 +20,52 @@ class ProposalType {
   // Filter by stage - stage can be one of: staged, active, done
   final String? filterByStage;
 
-  const ProposalType({
-    required this.index,
-    required this.type,
-    required this.lowerUpperBound,
-    required this.indexPosition,
-    required this.isReverse,
-    this.filterByStage
-  });
+  const ProposalType(
+      {required this.index,
+      required this.type,
+      required this.proposalStage,
+      required this.referendumStage,
+      required this.indexPosition,
+      required this.isReverse,
+      this.filterByStage});
 }
 
 const List<ProposalType> proposalTypes = [
   ProposalType(
     index: 0,
     type: 'Open',
-    lowerUpperBound: 'open',
+    proposalStage: 'open',
+    referendumStage: 'active',
     indexPosition: 2,
     isReverse: false,
-    filterByStage: "active"
+    filterByStage: 'active',
   ),
   ProposalType(
     index: 1,
     type: 'Upcoming',
-    lowerUpperBound: 'staged',
+    proposalStage: 'staged',
+    referendumStage: 'testing',
     indexPosition: 3,
     isReverse: false,
   ),
   ProposalType(
     index: 2,
     type: 'History',
-    lowerUpperBound: 'done',
+    proposalStage: 'done',
+    referendumStage: 'passed',
     indexPosition: 3,
     isReverse: true,
   ),
 ];
+
+// Proposals
+// 'Open': {'stage': 'active', 'status': 'open'},
+// 'Evaluate': {'stage': 'active', 'status': 'evaluate', 'reverse': 'true'},
+// 'Passed': {'stage': 'done', 'status': 'passed', 'reverse': 'true'},
+// 'Failed': {'stage': 'done', 'status': 'rejected', 'reverse': 'true'},
+
+// Referendums
+// 'Open': {'referendumStage': 'active'},
+// 'Evaluate': {'referendumStage': 'testing', 'reverse': 'true'},
+// 'Passed': { "referendumStage": "passed", 'reverse': 'true'},
+// 'Failed': { "referendumStage": "failed", 'reverse': 'true'},
```

Comment:
@n13  You said referendum uses stage as scope. But not sure which of these match the tabs of the application.
For example: 

OPEN -> active
UPCOMMING -> is this testing stage ?
HISTORY -> there doesn't seem to be just a done, there are 2 stages (passed and failed). Shoud I make 2 calls and then merge the results for done?

BTW I think we don't load the evaluate stage in any section.

Reply:
Open/Evaluate/Passed/Failed were the previous app categories

New app has ...?? 

Open/Upcoming/History

Open = Open
Upcoming - we did not have this before, the scope is called "staged"
History - passed + failed scopes

testing = evaluate stage, we don't show this

Referendums time line

Staged -> Open -> (Evaluate or failed) -> (passed or failed)

So open referendums when they "win" they go into evaluate; when they lose they go into failed. 

From evaluate, they can either go to passed or to failed.

Reply:
So add one more
```
'Upcoming': {'referendumStage': 'staged'},
```

This list should not be reversed.

Reply:
For History can I make a request to include both (passed or failed) or I must make a request for each one and then merge them?

Reply:
> @n13 For History..
> can I make a request to include both stages (passed or failed)?
> or
> I must make a request for each one and then merge them?

Two requests 

This is a limitation of the blockchain data query capabilities (which are primitive)